src/Table/Table.js

```diff
@@ -58,81 +68,149 @@ const Table = React.forwardRef(({ headers, tableData, className, compact, conden
+            gridManager.current.clearEvents();
+        } else {
+            attachGridManager(tableRef.current);
+        }
+    }, [keyboardNavigation]);
+
+    const captionId = useUniqueId();
     const gridManager = useRef(new GridManager());
-    const tableRef = ref || useHookWithRefCallback();
+    const tableRef = ref || useRef(null);
     const [instructionsText, setInstructionsText] = useState('');
 
+    const onKeyDownCell = (cell, event) => {
+        const key = event.which || event.keyCode;
+
+        if (keyboardNavigation === 'row' && event.target.matches(GridSelector.ROW) && key === keycode.codes.space) {
+            selection.onSelectRow(cell.row - 1);
+            event.preventDefault();
+            return;
+        }
+
+        setInstructionsText(generateInstructionsText(cell, event));
+    };
+
+    const onClickRowHandler = (cell) => {
+        selection?.onClickRow && selection.onClickRow(cell.row - 1);
+    };
+
     const onToggleEditMode = (enable) => {
         setInstructionsText(enable ? localizedText.editModeDisable : '');
     };
 
     const onFocusCell = (cell, event) => {
-        const { row, col } = cell;
-        const key = event.which || event.keyCode;
-        const navigatedHorizontally = (key === keycode.codes.left || key === keycode.codes.right) && col > 0;
-        const navigatedVertically = (key === keycode.codes.up || key === keycode.codes.down) && row > 0;
+        setInstructionsText(generateInstructionsText(cell, event));
+    };
+
+    const onBlurTable = (event) => {
+        if (event.target === tableRef.current) {
+            setInstructionsText('');
+        }
+    };
+
+    const generateInstructionsText = (cell, event) => {
         if (gridManager.current?.editMode) {
-            setInstructionsText(localizedText.editModeDisable);
-        } else {
-            let newInstructionsText = '';
-            if (navigatedVertically) {
-                newInstructionsText += `${localizedText.row} ${row} `;
-            } else if (navigatedHorizontally) {
-                newInstructionsText += `${localizedText.column} ${col} ${headers[col]} `;
-            } else {
-                newInstructionsText += `${localizedText.arrowKeys} `;
-            }
-            if (gridManager.current?.isEditableCell(cell)) {
-                newInstructionsText += localizedText.editModeEnable;
-            }
-            setInstructionsText(newInstructionsText);
+            return localizedText.editModeDisable;
         }
+
+        let newInstructionsText = '';
+
+        switch (event.type) {
+            case 'focus':
+                if (instructionsText.length === 0) {
+                    if (keyboardNavigation !== 'row') {
+                        newInstructionsText += `${localizedText.arrowKeys} `;
+                    } else {
+                        newInstructionsText += `${localizedText.rowArrowKeys} ${localizedText.rowSelection} ${localizedText.rowClick}`;
```

Comment:
Similar comment here about concatenating localized strings.

Reply:
Decided to combine the row instructions into a single prop to prevent concatenation.